Correctly enable user service

jsf9k opened the pull request:
## 🗣 Description ##

This pull request updates the Ansible role to enable the user `systemd` service in a way that works when creating AMIs and when testing via `molecule`.

## 💭 Motivation and context ##

These changes resolve #29.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
Correctly enable a user service

jsf9k committed:
Drop the molecule-idempotence-notest tag in favor of changed_when: false

This changed_when is necessary to pass idempotence.  We cannot use the
molecule-idempotence-notest tag here because the command must run.
See ansible/molecule#2765 for more details about this type of
situation.

jsf9k committed:
Prefer systemd_service to systemd

jsf9k committed:
Remove unnecessary, commented-out code

jsf9k committed:
Update comments to match recent changes

jsf9k committed:
Separate tasks with blank lines for readability

dav3r commented on tasks/enable_file_share_service.yml:
```diff
@@ -1,70 +1,51 @@
-    - molecule-idempotence-notest
+  # This changed_when is necessary to pass idempotence.  We cannot use
+  # the molecule-idempotence-notest tag here because the command must
+  # run; otherwise, the task the follows will fail.  See
```
Typo:
```suggestion
  # run; otherwise, the task that follows will fail.  See
```

jsf9k committed:
Fix typo

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Uncomment a test that now succeeds

Note that the test code required a little tweaking before it could
succeed due to pytest-dev/pytest-testinfra#766.

jsf9k committed:
Use the sudosu become method when enabling the user's systemd user service

Using su (together with sudo, since we are not sshing in as the root
user) to become the user causes an actual systemd session to be
created for the user.  This in turn causes the user's systemd user
services to be started, which we require for our Molecule tests.

Note that we also install the Python ansible package in our pre-commit
configuration.  This is necessary because the community.general.sudosu
become method is packaged there instead of in the ansible-core Python
package.

jsf9k committed:
Add comments to explain why the linger commands are run as root

jsf9k committed:
Prefer systemd_service to service when configuring a service provided by systemd

systemd is certainly the service manager to use in such a case.

jsf9k committed:
Improve comments to provide more or more accurate information

jsf9k committed:
Restart the systemd-logind service

Kali Linux, at least, currently has an issue where systemd-logind is
not restarted after an update and is left in a non-functional state.
This results in an inability to create a user systemd session, which
is necessary when interacting with systemd user services, so we force
a restart here.

dav3r commented:
@jsf9k You forgot to circle back and fill out the "Description" section of the PR description. 😄

dav3r reviewed (approved):
Strong work! 💪 🧠 💼
